core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -302,13 +323,29 @@ public Table loadTable(SessionContext context, TableIdentifier identifier) {
     }
 
     AuthSession session = tableSession(response.config(), session(context));
+    TableMetadata tableMetadata;
+
+    if (snapshotMode == SnapshotMode.REFS) {
+      tableMetadata =
+          TableMetadata.buildFrom(response.tableMetadata())
+              .setSnapshotsSupplier(
+                  () ->
+                      loadInternal(context, identifier, SnapshotMode.ALL)
+                          .tableMetadata()
+                          .snapshots())
+              .discardChanges()
```

Comment:
Are there any changes? This is probably just trying to be safe right?

Reply:
Yes, this is just to ensure that we aren't building anything that might get passed on in the metadata response.